src/sidebar/services/router.js

```diff
@@ -0,0 +1,85 @@
+import * as queryString from 'query-string';
+
+/**
+ * A service that manages the association between the route and route parameters
+ * implied by the URL and the corresponding route state in the store.
+ */
+// @ngInject
+export default function router($window, store) {
+  /**
+   * Return the name and parameters of the current route.
+   */
+  function currentRoute() {
+    const path = $window.location.pathname;
+    const pathSegments = path.slice(1).split('/');
+    const params = queryString.parse($window.location.search);
+
+    let route;
+    if (pathSegments[0] === 'a') {
+      route = 'annotation';
+      params.id = pathSegments[1] || '';
+    } else if (pathSegments[0] === 'stream') {
+      route = 'stream';
+    } else {
+      route = 'sidebar';
+    }
+
+    return { route, params };
+  }
+
+  /**
+   * Generate a URL for a given route.
+   */
+  function routeUrl(name, params = {}) {
+    let url;
+    const queryParams = { ...params };
+
+    if (name === 'annotation') {
```

Comment:
Done. I didn't intend this code to be that extensible because I figure if we're adding a bunch more routes in future then we'll probably add a proper router to map between URL patterns and route names/params.